extensions/positron-python/python_files/posit/positron/variables.py

```diff
@@ -707,6 +712,96 @@ def _send_details(self, _path: list[str], value: Any = None):
         msg = InspectedVariable(children=children, length=len(children))
         self._send_result(msg.dict())
 
+    def _perform_get_variable_summary(self, path: list[str]) -> None:
+        """RPC handler for getting variable data summary."""
+        try:
+            self._summarize_data(path)
+        except Exception as err:
+            self._send_error(
+                JsonRpcErrorCode.INTERNAL_ERROR,
+                f"Error summarizing variable at '{path}': {err}",
+            )
+
+    def _summarize_data(self, path: list[str]):
+        """Compute statistical summary for a variable without opening a data explorer."""
+        from .data_explorer import DataExplorerState, _get_table_view, _value_type_is_supported
+        from .data_explorer_comm import FormatOptions
+
+        is_found, value = self._find_var(path)
+        if not is_found:
+            raise ValueError(f"Cannot find variable at '{path}' to summarize")
+
+        if not _value_type_is_supported(value):
+            raise ValueError(f"Variable at '{path}' is not supported for summary")
+
+        try:
+            # Create a temporary table view with a temporary comm
+            temp_state = DataExplorerState("temp_summary")
+            temp_comm = PositronComm.create(target_name="temp_summary", comm_id="temp_summary_comm")
+            table_view = _get_table_view(value, temp_comm, temp_state, self.kernel.job_queue)
+        except Exception as e:
+            raise ValueError(f"Failed to create table view: {e}") from e
+
+        # Get the number of columns and build schema manually
+        try:
+            num_columns = table_view.table.shape[1]
+            num_rows = table_view.table.shape[0]
+
+            # Get column schemas directly using the internal method
+            column_schemas = []
+            for i in range(num_columns):
+                column_schema = table_view._get_single_column_schema(i)  # noqa: SLF001
+                column_schemas.append(column_schema)
+
+            # Create schema object manually
+            from .data_explorer_comm import TableSchema
+
+            schema = TableSchema(columns=column_schemas)
+        except Exception as e:
+            raise ValueError(f"Failed to get schema: {e}") from e
```

Comment:
We should be able to push some of these tool-calling helpers into functions in data_explorer.py that we can unit test there

Reply:
I haven't added any unit tests, but I did add `_get_column_profiles` and `_get_table_schema_from_view` as helpers to data_explorer.py